packages/core-plugin-api/src/analytics/useAnalytics.tsx

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useAnalyticsDomain } from './AnalyticsDomain';
+import {
+  analyticsApiRef,
+  AnalyticsTracker,
+} from '../apis/definitions/AnalyticsApi';
+import { useApi } from '../apis';
+
+function useTracker(): AnalyticsTracker {
+  const analyticsApi = useApi(analyticsApiRef);
+  const domain = useAnalyticsDomain();
+  return {
```

Comment:
Should `useMemo` this to make sure that we don't need to return a new instance too often. Another option that might be nicer is to create a small class that does this wrapping and then store an instance of that here with a `useRef`. You'd then have some method on the class that we can use to mutate in as the `domain` changes, keeping the returned `AnalyticsTracker` reference completely stable.